yt_dlp/extractor/archiveorg.py

```diff
@@ -241,3 +251,87 @@ def _real_extract(self, url):
                     'parent': 'root'})
 
         return info
+
+
+class YoutubeWebArchiveIE(InfoExtractor):
+    IE_NAME = 'youtube:web.archive.org'
+    IE_DESC = 'web.archive.org saved youtube videos'
+    _VALID_URL = r"""(?x)^
+                (?:https?://|//)?web\.archive\.org\/
+                    (?:web/)?
+                    (?:[0-9A-Za-z_*]+/)? # /web and the version index is optional
+                (?:https?:\/\/)?
+                (?:
+                    (?:\w+\.)?youtube\.com\/watch\?v= # Youtube URL
+                    |(wayback-fakeurl\.archive\.org\/yt\/) # Or optionally, also support the internal fake url
```

Comment:
You might also want to deal with encoded urls here

Reply:
We could possibly bring in some of the regex from the youtube extractor for this part, but not 100% sure on what.

Reply:
the youtube _VALID_URL is very complex in order to deal with the url from various other related sites (like invidious) and embed urls from various popular websites. Most of that is not necessary here